### ISSUE TITLE
Higher-level API with node-style callbacks or promises

### ISSUE THREAD

**Issue by ipoddubny** [ipoddubny/ya-node-asterisk#3]
One AMI request can lead to a series of replies from Asterisk, each calling the callback set by user.
It would be very nice to have a higher level API that calls the user callback **only once**, after having collected all the replies from Asterisk.

**Comment by avesus** [comment]
I use this function to issue AMI commands:
```js
function sendAsync(input) {
  var ami = this;
  return new Promise(function(fulfill, reject) {
    var eventList = [];
    var fetchingEvents = false;
    ami.send(input, function(res) {
      if (!fetchingEvents) {
        if (res.Response === 'Error') {
          reject(res.Message);
        }
        else if (res.EventList === 'start') {
          fetchingEvents = true;
        }
        else if (res.Response === 'Success') {
          if (res.Message !== 'Queue status will follow') {
            fulfill(res);
          } else {
            fetchingEvents = true;
          }
        } else {
          console.log(res);
        }
      }
      else if (res.EventList === 'Complete') {
        fulfill(eventList);
      }
      else if (res.Event) {
        if (res.Event === 'QueueStatusComplete') {
          fulfill(eventList);
        } else if (res.Event !== 'QueueMember' && res.Event !== 'QueueParams') {
          eventList.push(res);
        }
      } else {
        console.log(res);
      }
    });
  });
};
```

**Comment by ipoddubny** [comment]
The only way to do it properly is to explicitly list all the available actions and use action-specific handlers, because even in Asterisk 13 AMI is still inconsistent. Some actions use event lists, while others don't.
